npm/pkg/controlplane/translation/translatePolicy.go

```diff
@@ -472,56 +505,82 @@ func isAllowAllToEgress(egress []networkingv1.NetworkPolicyEgressRule) bool {
 	if isAllowAllToEgress(egress) {
 		allowAllPolicy(npmNetPol, policies.Egress)
-		return
+		return nil
 	}
 
 	// #2. If egress is nil (in yaml file, it is specified with '[]'), it means "Deny all" - it does not allow sending traffic to others.
 	if egress != nil {
 		// Except for allow all traffic case in #1, the rest of them should have default drop rules.
 		dropACL := defaultDropACL(npmNetPol.NameSpace, npmNetPol.Name, policies.Egress)
 		npmNetPol.ACLs = append(npmNetPol.ACLs, dropACL)
-		return
+		return nil
 	}
 
 	// #3. Egress rule is not AllowAll (including internal and external) and DenyAll.
 	// So, start translating egress policy.
 	for i, rule := range egress {
-		translateRule(npmNetPol, policies.Egress, policies.DstMatch, i, rule.Ports, rule.To)
+		err := translateRule(npmNetPol, policies.Egress, policies.DstMatch, i, rule.Ports, rule.To)
+		if err != nil {
+			return err
+		}
 	}
 
 	// #3. Except for allow all traffic case in #1, the rest of them should have default drop rules.
 	// Add drop ACL to drop the rest of traffic which is not specified in Egress Spec.
 	dropACL := defaultDropACL(npmNetPol.NameSpace, npmNetPol.Name, policies.Egress)
 	npmNetPol.ACLs = append(npmNetPol.ACLs, dropACL)
+	return nil
 }
 
 // TranslatePolicy traslates networkpolicy object to NPMNetworkPolicy object
 // and return the NPMNetworkPolicy object.
-func TranslatePolicy(npObj *networkingv1.NetworkPolicy) *policies.NPMNetworkPolicy {
+func TranslatePolicy(npObj *networkingv1.NetworkPolicy) (*policies.NPMNetworkPolicy, error) {
 	npmNetPol := &policies.NPMNetworkPolicy{
 		Name:      npObj.ObjectMeta.Name,
 		NameSpace: npObj.ObjectMeta.Namespace,
 	}
 
 	// podSelector in spec.PodSelector is common for ingress and egress.
 	// Process this podSelector first.
-	npmNetPol.PodSelectorIPSets, npmNetPol.PodSelectorList = podSelectorWithNS(npmNetPol.NameSpace, policies.EitherMatch, &npObj.Spec.PodSelector)
+	var err error
+	npmNetPol.PodSelectorIPSets, npmNetPol.PodSelectorList, err = podSelectorWithNS(npmNetPol.NameSpace, policies.EitherMatch, &npObj.Spec.PodSelector)
+	if err != nil {
+		return nil, err
+	}
 
 	// Each NetworkPolicy includes a policyTypes list which may include either Ingress, Egress, or both.
 	// If no policyTypes are specified on a NetworkPolicy then by default Ingress will always be set
 	// and Egress will be set if the NetworkPolicy has any egress rules.
 	for _, ptype := range npObj.Spec.PolicyTypes {
 		if ptype == networkingv1.PolicyTypeIngress {
-			ingressPolicy(npmNetPol, npObj.Spec.Ingress)
+			err := ingressPolicy(npmNetPol, npObj.Spec.Ingress)
+			if err != nil {
+				return nil, err
+			}
 		} else {
-			egressPolicy(npmNetPol, npObj.Spec.Egress)
+			err := egressPolicy(npmNetPol, npObj.Spec.Egress)
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
 
-	return npmNetPol
+	return npmNetPol, nil
+}
+
+func invalidWindowsOperatorLimitation(op metav1.LabelSelectorOperator) bool {
```

Comment:
nitpick: looks like we just use this in parseSelector.go. Can we put it in that file?